### PR TITLE
Fixes strange geyser and geyser regen

### LINKED ISSUE #72037: Strange Geysers not working
## Round ID:
N/A

## Testmerges:
N/A

## Reproduction:
discover any strange geyser on lavaland with a mining scanner, they all make Oil now, which is unintended as far as I know. After the change where Geysers didn't need to be plungered, a few days past that, all the strange geysers stopped being random and only make Oil instead of a random chemical. Thought it would be fun to run the 'Holy shit theres oil here' for awhile but honestly, its not nearly as fun as finding a geyser that produces condensed capsaicin or hot ramen on an ice planet. Plus the lavaland ruin with the graves already had oil geysers, so theres that.

### PR DESCRIPTION
**Pull request by Time-Green**
Called the random reagent code after it was initialized
Registered the reagent del/remove signal on the geyser, not the reagent datum of the geyser

Closes #72037 

:cl:
fix: Strange geysers have random reagents again
fix: Geysers regen reagents again
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
you should add unit test

**Comment by Time-Green** [comment]
Yeah yeah I’ll get to it

**Comment by Time-Green** [comment]
I got to it

File: code/modules/unit_tests/geyser.dm [hunk 1]
```diff
@@ -0,0 +1,27 @@
+///Geysers listen to reagent signals to know when to start processing, which is very cool, smart, optimized and fragile
```
Comment (Mothblocks) [inline code/modules/unit_tests/geyser.dm:1]: You should explain better waht this is actually testing

File: code/modules/unit_tests/geyser.dm [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+/datum/unit_test/geyser/Run()
```
Comment (Mothblocks) [inline code/modules/unit_tests/geyser.dm:2]:
```suggestion
/datum/unit_test/geyser

/datum/unit_test/geyser/Run()
```

We want to document the test, not `Run()`

File: code/modules/unit_tests/geyser.dm [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+		TEST_ASSERT(geyser.reagents, "Geyser does not have a reagent datum! Source: [geyser.type]")
+		TEST_ASSERT(geyser.reagents.has_reagent(geyser.reagent_id), "Geyser should start with [geyser.reagent_id], but started with [geyser.reagents.get_reagent_log_string()] instead. Source: [geyser.type]")
+
+		geyser.reagents.clear_reagents() //this should awaken the geyser to start refilling
+
+		var/do_we_do_autorefill = FALSE
+		var/time_passed = 0 SECONDS
+		//shit can be delayed due to laggines or something, so check every 2 seconds. if it lasts longer than 30 somethings definitely fucking wrong
+		while(time_passed < 30 SECONDS)
```
Comment (Mothblocks) [inline code/modules/unit_tests/geyser.dm:17]: 1. You have to change the priority of the test, or you hold up smaller tests which makes running tests locally more annoying
2. You already shouldn't really be sleeping in tests, but 30 seconds is definitely too high IMO. I'm not thrilled with how long our tests are but the ones that are long catch very common regressions, this is definitely past my limit. What are we doing here?

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Mothblocks** [comment]
Ideally the way you do time based unit tests is to create subtypes/traits for things to say "do this significantly faster"

**Comment by Time-Green** [comment]
> Ideally the way you do time based unit tests is to create subtypes/traits for things to say "do this significantly faster"

Aargh thats what I did at first, okay

**Comment by Time-Green** [comment]
@Mothblocks I took the easy way out and now just fire the subsystem directly to skip waiting completely

**Pull request merged**